File: Daily Challenge/February 2025/Maximum Absolute Sum of Any Subarray.cs
public class Solution {
  public int MaxAbsoluteSum(int[] nums) {
    int maxSum = MaxSubarraySum(nums);
    for (int i = 0; i < nums.Length; ++i) {
      nums[i] = -nums[i];
    }

    return Math.Max(maxSum, MaxSubarraySum(nums));
  }

  private int MaxSubarraySum(int[] nums) {
    int maxSum = int.MinValue;
    int minSum = 0;
    int sum = 0;

    foreach (int num in nums) {
      sum += num;
      maxSum = Math.Max(maxSum, sum - minSum);
      minSum = Math.Min(minSum, sum);
    }

    return Math.Abs(maxSum);
  }
}

File: Daily Challenge/February 2025/Maximum Absolute Sum of Any Subarray.cpp
class Solution {
public:
  int maxAbsoluteSum(vector<int>& nums) {
    int maxSum = maxSubarraySum(nums);
    for (int& num : nums) {
      num *= -1;
    }
    
    return max(maxSum, maxSubarraySum(nums));
  }

private:
  int maxSubarraySum(vector<int>& nums) {
    int maxSum = INT_MIN;
    int minSum = 0;
    int sum = 0;

    for (int num : nums) {
      sum += num;
      maxSum = max(maxSum, sum - minSum);
      minSum = min(minSum, sum);
    }

    return abs(maxSum);
  }
};